addons/mail/static/src/components/discuss/discuss.xml

```diff
@@ -60,6 +60,7 @@
                 hasSquashCloseMessages="discuss.thread.model !== 'mail.box'"
                 haveMessagesMarkAsReadIcon="discuss.thread === env.messaging.inbox"
                 haveMessagesReplyIcon="discuss.thread === env.messaging.inbox"
+                haveMessagesUnfollowIcon="discuss.thread.model === 'mail.box'"
```

Comment:
This should be specific to inbox like the others above. `mail.box` includes other stuff on which the button makes no sense (eg. moderation).